backend/v3/storage/database/events_testing/settings_instance_test.go

```diff
@@ -0,0 +1,1790 @@
+			AllowUsernamePassword:      false,
+			AllowRegister:              false,
+			AllowExternalIdp:           true,
+			ForceMfa:                   true,
+			PasswordlessType:           policy.PasswordlessType_PASSWORDLESS_TYPE_NOT_ALLOWED,
+			HidePasswordReset:          true,
+			IgnoreUnknownUsernames:     true,
+			DefaultRedirectUri:         "http://www.example.com",
+			PasswordCheckLifetime:      durationpb.New(time.Second * 20 * 20),
+			ExternalLoginCheckLifetime: durationpb.New(time.Second * 20 * 21),
+			MfaInitSkipLifetime:        durationpb.New(time.Second * 20 * 22),
+			SecondFactorCheckLifetime:  durationpb.New(time.Second * 20 * 23),
+			MultiFactorCheckLifetime:   durationpb.New(time.Second * 20 * 24),
+			AllowDomainDiscovery:       false,
+			DisableLoginWithEmail:      true,
+			DisableLoginWithPhone:      true,
+			ForceMfaLocalOnly:          true,
+		})
+		require.NoError(t, err)
+		after := time.Now()
+
+		retryDuration, tick := integration.WaitForAndTickWithMaxDuration(ctx, time.Second*20)
+		assert.EventuallyWithT(t, func(t *assert.CollectT) {
+			setting, err := settingsRepo.Get(
+				ctx, pool,
+				database.WithCondition(
+					database.And(
+						settingsRepo.InstanceIDCondition(newInstance.ID()),
+						settingsRepo.OrganizationIDCondition(nil),
+						settingsRepo.TypeCondition(domain.SettingTypeLogin),
+						settingsRepo.OwnerTypeCondition(domain.OwnerTypeInstance),
+					),
+				),
+			)
+			require.NoError(t, err)
+
+			require.NotNil(t, setting.ForceMFA)
+
+			// event instance.policy.login.changed
+			assert.Equal(t, domain.OwnerTypeInstance, setting.OwnerType)
+			assert.Equal(t, false, setting.AllowRegister)
+			assert.Equal(t, true, setting.AllowExternalIDP)
+			assert.Equal(t, true, setting.ForceMFA)
+			assert.Equal(t, domain.PasswordlessTypeNotAllowed, setting.PasswordlessType)
+			assert.Equal(t, true, setting.HidePasswordReset)
+			assert.Equal(t, true, setting.IgnoreUnknownUsernames)
+			assert.Equal(t, "http://www.example.com", setting.DefaultRedirectURI)
+			assert.Equal(t, false, setting.AllowDomainDiscovery)
+			assert.Equal(t, false, setting.AllowUserNamePassword)
+			assert.Equal(t, time.Duration(time.Second*20*20), setting.PasswordCheckLifetime)
+			assert.Equal(t, time.Duration(time.Second*20*21), setting.ExternalLoginCheckLifetime)
+			assert.Equal(t, time.Duration(time.Second*20*22), setting.MFAInitSkipLifetime)
+			assert.Equal(t, time.Duration(time.Second*20*23), setting.SecondFactorCheckLifetime)
+			assert.Equal(t, time.Duration(time.Second*20*24), setting.MultiFactorCheckLifetime)
+			assert.Equal(t, true, setting.DisableLoginWithEmail)
+			assert.Equal(t, true, setting.DisableLoginWithPhone)
+			assert.Equal(t, true, setting.ForceMFALocalOnly)
+			assert.WithinRange(t, *setting.UpdatedAt, before, after)
+		}, retryDuration, tick)
+	})
+
+	t.Run("test added/remove login multifactor type reduces", func(t *testing.T) {
+		ctx := t.Context()
```

Comment:
Why do you define ctx, but then a line later still use t.Context() directly? I would just always use t.Context().